plugins/available/goenv.plugin.bash

```diff
@@ -0,0 +1,21 @@
+cite about-plugin
+about-plugin 'Init goenv, if installed. Plays nicely with package managers'
+
+# Try to set GOENV_ROOT if not already set
+if [ -z "${GOENV_ROOT}" ]; then
+	# Bail if cannot be set
+	if [ ! -d "$HOME/.goenv" ]; then
+		return
+	fi
+	export GOENV_ROOT="$HOME/.goenv"
+fi
+
+# Add GOENV_ROOT/bin to path if goenv not already on path
+if ! _command_exists goenv && [ -x "${GOENV_ROOT}/bin/goenv" ]; then
+	pathmunge "$GOENV_ROOT/bin"
+fi
+
+# Initialize goenv - Play nicely if goenv not on path
+if _command_exists goenv ; then
```

Comment:
As you pointed out in #1510, this is a poor check if you are worried about double initialization. Instead `$(type -t goenv) == 'function'` would be more accurate.

Reply:
Hi ... As much as I wouldn't mind avoiding double-initializing, there is no *safe* method for doing so.

What I mean by *safe* is that the results of the eval are private to the tool and not something we sould be checking or basing logic on ... I have not seen anything in there docs regarding a variable or state we could trust (i.e. documented as public) to check for previous initialization. 

Thats why I mentioned in my notes that I preferred the GOENV_ROOT check over the function check, but then you showed me that the GOENV_ROOT check was wrong, so that leaves me with preferring no check.

*UPDATE*: I just tried testing the double-initialization and there were no issues.

 The function simply delegates to `command goenv` which executes the command directly.

This appears to safely re-initialize the function and other variables.

So I think we're good